### DISCUSSION TITLE
Support PS5 controller (DualSense)

### DISCUSSION THREAD

**Discussion by o0Zz** [o0Zz/sys-con#13] [Q&A]
Does sys-con 1.x support PS5 Controller ?
If anyone has a PS5 controller to test I'm interrested to know.
After a quick look the PS5 expose a generic HID profile which means sys-con should natively support it.

https://github.com/nondebug/dualsense

Thanks

**Comment by o0Zz** [comment]
@onLook3rEk0r I open a dedicated discussion for this topic. Let me know the result with your controller

**Reply by onLook3rEk0r** [reply]
It works, I tried

**Reply by onLook3rEk0r** [reply]
But it seems like a huge input delay

**Reply by o0Zz** [reply]
What is your log level ? Did you change it ?

**Reply by o0Zz** [reply]
Could you share the binding you set in config.ini for other players ;) ?

**Reply by onLook3rEk0r** [reply]
I did not change the log level, I just add the vid&pid of my controller and lower the deadzone to 0.

**Reply by o0Zz** [reply]
Did you bind the buttons ?

**Reply by onLook3rEk0r** [reply]
Nope, I use the profile of dualshock4

**Reply by onLook3rEk0r** [reply]
Default as ur settings.

**Reply by o0Zz** [reply]
And all buttons match their functions ?

**Reply by o0Zz** [reply]
Could you share the [vid-pid] please

**Reply by onLook3rEk0r** [reply]
One of my friend said the part of right of his dualsense controller (ABXY) can not match. Is there some issue ?

**Reply by onLook3rEk0r** [reply]
Dualsense (PS5) [054c-0ce6]   Dualsense Edge[054c-0df2]

**Reply by onLook3rEk0r** [reply]
> One of my friend said the part of right of his dualsense controller (ABXY) can not match. Is there some issue ?

He said it goes well at the first time, but the button suddenly can not use in somehow

**Comment by o0Zz** [comment]
Follow the documentation ->[README](https://github.com/o0Zz/sys-con/?tab=readme-ov-file#how-to-add-a-new-controller-) 
You need to map your buttons:
```
B=2
A=3
Y=1
X=4
...
```

Once you found the correct mapping, I'm interressted to have it, if you can share it

**Reply by onLook3rEk0r** [reply]
Yes, I’ll test it when I come home, and share it later. Thx for ur code sincerely. And do u have Dualsense or DSE controller by ur side?

**Reply by onLook3rEk0r** [reply]
this can't be supported in Atmosphere 1.7.1?

**Reply by o0Zz** [reply]
Did you try to sys-module atmosphere 1.7.0 (It suppose to work with any atmosphere 1.7.x) ? Did you get a crash or anything ?

**Reply by texita** [reply]
> Did you try to sys-module atmosphere 1.7.0 (It suppose to work with any atmosphere 1.7.x) ? Did you get a crash or anything ?

i'm using sys-con 1.1.1 in atmosphere 1.7.1, so yes, it's working.

**Reply by onLook3rEk0r** [reply]
> > Did you try to sys-module atmosphere 1.7.0 (It suppose to work with any atmosphere 1.7.x) ? Did you get a crash or anything ?
> 
> i'm using sys-con 1.1.1 in atmosphere 1.7.1, so yes, it's working.

how? my sys-con always crash when my system is boosting. It is also atmosphere 1.7.1

**Reply by onLook3rEk0r** [reply]
> Did you try to sys-module atmosphere 1.7.0 (It suppose to work with any atmosphere 1.7.x) ? Did you get a crash or anything ?

A crash from atmosphere/contents/690000000D

**Reply by onLook3rEk0r** [reply]
> Did you try to sys-module atmosphere 1.7.0 (It suppose to work with any atmosphere 1.7.x) ? Did you get a crash or anything ?

So I degraded my system to 18.0.1 & atmosphere to 1.7.0, it is finally work.

**Reply by onLook3rEk0r** [reply]
> > > Did you try to sys-module atmosphere 1.7.0 (It suppose to work with any atmosphere 1.7.x) ? Did you get a crash or anything ?
> > 
> > 
> > i'm using sys-con 1.1.1 in atmosphere 1.7.1, so yes, it's working.
> 
> how? my sys-con always crash when my system is boosting. It is also atmosphere 1.7.1

Boot

**Reply by o0Zz** [reply]
Thius issue report the same problem: https://github.com/o0Zz/sys-con/issues/15  (You could try to binary provided, it should fix the issue)

**Reply by o0Zz** [reply]
BTW, did you progress on the PS5 binding ?

**Reply by onLook3rEk0r** [reply]
> BTW, did you progress on the PS5 binding ?

I have submited it in ur forks
